### PR TITLE
3 Cycle

### PR DESCRIPTION
**Pull request by ishwari20**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given a graph with N vertices (numbered from 0 to N-1) and M undirected edges, then count the distinct 3-cycles in the graph. A 3-cycle PQR is a cycle in which (P,Q), (Q,R) and (R,P) are connected by an edge.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #4820

### CODE REVIEW

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Shift your code to some other function and use main for driver code.

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -198,10 +198,7 @@
 - [Floor value of a given number in the given BST](ds/Floor_of_a_number_in_BST.cpp)
 - [Print all node values of a given BST, which lies within the given range](ds\BST_Range_Print.cpp)
 - [Check if two given trees are Quasi Isomorphic or not](ds/Quasi_Isomorphic_Binary_Trees.cpp)
-- [Build a Balanced BST with given vector of values](ds/Balanced_BST_From_a_vector_of_values.cpp)
-- [Given a BST check if it is AVL Tree or not](ds/AVL_Tree_or_not.cpp)
-- [Searching in BST](ds/Searching_in_BST.cpp)
-- [Binary tree using stack (iterative method to construct tree)](ds/BinaryTreeUsingStack.cpp)
```
Comment (siddharth25pandey) [inline C-Plus-Plus/README.md:204]: Add the unnecessary deleted lines

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/*
+Given a graph with N vertices (numbered from 0 to N-1) and M undirected edges, then count the distinct
+3-cycles in the graph. A 3-cycle PQR is a cycle in which (P,Q), (Q,R) and (R,P) are connected by an edge.
+*/ 
+
+#include <iostream>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:6]: Use **#include<bits/stdc++.h>**

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+              }
+          }
+        }
+    }
+    cout<< count/6;
+}
+
+int main() {
+    
+    int V,E;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:30]: Use good naming convention

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    cin>>V>>E;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:31]: Display message in Print form as what are you taking input from the user

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    int** edges=new int*[V];
+    for(int i=0;i<V;i++){
+        edges[i]=new int[V];
+        for(int j=0;j<V;j++){
+            edges[i][j]=0;
+        }
+    }
+    for(int i=0;i<E;i++){
+        int f,s;
+        cin>>f>>s;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:41]: Display message in Print form as what are you taking input from the user

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    cin>>V>>E;
+    int** edges=new int*[V];
+    for(int i=0;i<V;i++){
+        edges[i]=new int[V];
+        for(int j=0;j<V;j++){
+            edges[i][j]=0;
+        }
+    }
+    for(int i=0;i<E;i++){
+        int f,s;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:40]: Use good naming convention

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        cin>>f>>s;
+        edges[f][s]=1;
+        edges[s][f]=1;
+    }
+    count(V, edges);
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:45]: Display message in Print form as what are you showing output to the user

File: C-Plus-Plus/graphs/3Cycle.cpp [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+/*
+Sample Input 1:
+  3 3
+  0 1
+  1 2
+  2 0
+Sample Output 1:
+  1
+  
+Time Complexity: O(V+E)
```
Comment (siddharth25pandey) [inline C-Plus-Plus/graphs/3Cycle.cpp:59]: Also, include Space Complexity

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ishwari20 Do the requested changes

**Pull request closed without merging**